utils/managed_bytes.cc

```diff
@@ -23,8 +23,44 @@
 
 #include "managed_bytes.hh"
 
-managed_bytes::managed_bytes(managed_bytes_view o) : managed_bytes(initialized_later(), o.size()) {
-    // FIXME: implement
+managed_bytes::managed_bytes(const managed_bytes_view o) : managed_bytes(initialized_later(), o.size()) {
+    auto it = o.begin();
+    if (!external()) {
+        auto dst = _u.small.data;
+        while (it != o.end()) {
+            auto n = it->size();
+            memcpy(dst, it->data(), n);
+            dst += n;
+            it.remove_prefix(n);
+        }
+        return;
+    }
+    auto b = _u.ptr;
+    auto dst = b->data;
+    size_t dst_size = b->frag_size;
+    auto advance_dst = [&] (size_t n) {
+        if (n < dst_size) {
+            dst += n;
+            dst_size -= n;
+        } else {
+            assert(n == dst_size);
+            b = b->next;
+            if (b) {
+                dst = b->data;
+                dst_size = b->frag_size;
+            } else {
+                dst = nullptr;
+                dst_size = 0;
+            }
+        }
+    };
+    while (it != o.end()) {
+        auto n = std::min(it->size(), dst_size);
+        memcpy(dst, it->data(), n);
+        advance_dst(n);
+        it.remove_prefix(n);
+    }
+    assert(!b);
```

Comment:
Don't we already have this code in the `managed_bytes` copy constructor? We should reuse it.

Reply:
They are different, as the former copies `managed_bytes` and this one copies a `managed_bytes_view`.
Maybe if we change the former to use the `fragment_iterator` we could use common code.

Reply:
I think that would be best.